clients/include/testing_hpr_strided_batched.hpp

```diff
@@ -48,74 +47,106 @@ hipblasStatus_t testing_hpr_strided_batched(const Arguments& argus)
+    CHECK_HIP_ERROR(hipMemcpy(dA, hA.data(), sizeof(T) * A_size, hipMemcpyHostToDevice));
 
-        if(status != HIPBLAS_STATUS_SUCCESS)
-        {
-            hipblasDestroy(handle);
-            return status;
-        }
-    }
+    CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_DEVICE));
+    CHECK_HIPBLAS_ERROR(hipblasHprStridedBatchedFn(
+        handle, uplo, N, d_alpha, dx, incx, stride_x, dA, stride_A, batch_count));
 
-    // copy output from device to CPU
-    hipMemcpy(hA.data(), dA, sizeof(T) * A_size, hipMemcpyDeviceToHost);
+    CHECK_HIP_ERROR(hipMemcpy(hA_device.data(), dA, sizeof(T) * A_size, hipMemcpyDeviceToHost));
 
-    if(argus.unit_check)
+    if(argus.unit_check || argus.norm_check)
     {
         /* =====================================================================
            CPU BLAS
         =================================================================== */
         for(int b = 0; b < batch_count; b++)
         {
-            cblas_hpr<T>(uplo, N, alpha, hx.data() + b * stride_x, incx, hB.data() + b * stride_A);
+            cblas_hpr<T>(
+                uplo, N, h_alpha, hx.data() + b * stride_x, incx, hA_cpu.data() + b * stride_A);
         }
 
         // enable unit check, notice unit check is not invasive, but norm check is,
         // unit check and norm check can not be interchanged their order
         if(argus.unit_check)
         {
-            unit_check_general<T>(1, dim_A, batch_count, 1, stride_A, hB.data(), hA.data());
+            unit_check_general<T>(
+                1, dim_A, batch_count, 1, stride_A, hA_cpu.data(), hA_host.data());
+            unit_check_general<T>(
+                1, dim_A, batch_count, 1, stride_A, hA_cpu.data(), hA_device.data());
+        }
+        if(argus.norm_check)
+        {
+            hipblas_error_host = norm_check_general<T>(
+                'F', 1, dim_A, 1, stride_A, hA_cpu.data(), hA_host.data(), batch_count);
+            hipblas_error_device = norm_check_general<T>(
+                'F', 1, dim_A, 1, stride_A, hA_cpu.data(), hA_device.data(), batch_count);
+        }
+    }
+
+    if(argus.timing)
+    {
+        CHECK_HIP_ERROR(hipMemcpy(dA, hA.data(), sizeof(T) * A_size, hipMemcpyHostToDevice));
+        hipStream_t stream;
+        CHECK_HIPBLAS_ERROR(hipblasGetStream(handle, &stream));
+        CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_DEVICE));
+
+        int runs = argus.cold_iters + argus.iters;
+        for(int iter = 0; iter < runs; iter++)
+        {
+            if(iter == argus.cold_iters)
+                gpu_time_used = get_time_us_sync(stream);
+
+            CHECK_HIPBLAS_ERROR(hipblasHprStridedBatchedFn(
+                handle, uplo, N, d_alpha, dx, incx, stride_x, dA, stride_A, batch_count));
         }
+        gpu_time_used = get_time_us_sync(stream) - gpu_time_used;
+
+        ArgumentModel<e_N, e_incx, e_stride_x, e_stride_a, e_batch_count>{}.log_args<T>(
```

Comment:
Double check if the ` log_args<T>`  should be U for alpha logging (which could be logged) as the GLOPs and GBYTES already get T separately.